lib/liquid/partial_cache.rb

```diff
@@ -0,0 +1,18 @@
+module Liquid
+  class PartialCache
+    def self.load(template_name, context:, parse_context:)
+      cached_partials = (context.registers[:cached_partials] ||= {})
+      cached = cached_partials[template_name]
+      return cached if cached
+
+      file_system = (context.registers[:file_system] ||= Liquid::Template.file_system)
+      source = file_system.read_template_file(template_name)
+      parse_context.partial = true
+      Liquid::Template.parse(source, parse_context).tap do |partial|
```

Comment:
Nit: since this is a hot code path I would avoid the use of `tap`

Reply:
This actually ended up even cleaner since assignment returns the value anyway 🙌